Extensions/fonts/xrender-fonts.lisp

```diff
@@ -348,8 +350,16 @@
   (options nil)
   (device-name nil))
 
-(defmethod clim-clx::text-style-to-X-font :around
-    ((port clim-clx::clx-port) (text-style climi::device-font-text-style))
+;; Why are we getting an undefined class in the DEFMETHOD of
+;; CLIM-CLX:PORT-FIND-ALL-FONT-FAMILIES if we don't have this
+;; EVAL-WHEN here?
+(eval-when (:compile-toplevel :load-toplevel :execute)
```

Comment:
compilation order maybe? Either way wrapping it in eval-when is not a solution – it only masks the underlying problem.

Reply:
The only alternative solution I can think of is to put the class definition in its own file. Is that a better solution?

I think the root cause is that `DEFCLASS` expands to an `(EVAL-WHEN (:COMPILE-TOPLEVEL) ...)` form, which means that the class will not be available in macroexpand functions when the fasl is loaded.